### PR TITLE
Added code coverage using mysql

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
Currently running using mysql 5.7, after https://github.com/owncloud/core/pull/29409 is merged it will run agains mysql 5.5.

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
codecov shows that mysql code paths are touched - nice - https://codecov.io/gh/owncloud/core/pull/29410/changes

Adding --group DB would give us some speed up here as well ...

**Comment by SergioBertolinSG** [comment]
> Adding --group DB would give us some speed up here as well ...

I've added it in the last commit, is that ok?

File: tests/drone/test-coverage.sh [hunk 1]
```diff
@@ -44,5 +44,11 @@ fi
 ./occ app:enable federation
 ./occ app:enable federatedfilesharing
 
+if [[ "${DB_TYPE}" == "none" ]]; then
+  GROUP="--exclude-group DB"
+else
+  GROUP="--group DB"
```
Comment (DeepDiver1975) [inline tests/drone/test-coverage.sh]: now sqlite is executed with --group DB as well - DB_TYPE none is missing

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

**Comment by SergioBertolinSG** [comment]
Some failures in drone after last commit:

<details>
<summary>1) Test\Encryption\EncryptionWrapperTest::testWrapStorage with data set #0 (true, array())</summary>

Your test case is not allowed to access the database.


/drone/src/tests/lib/TestCase.php:110
/drone/src/lib/composer/pimple/pimple/src/Pimple/Container.php:113
/drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:111
/drone/src/lib/private/ServerContainer.php:87
/drone/src/lib/private/Server.php:1141
/drone/src/lib/private/AllConfig.php:90
/drone/src/lib/private/AllConfig.php:340
/drone/src/lib/private/AllConfig.php:251
/drone/src/lib/private/legacy/util.php:233
/drone/src/lib/private/Files/Filesystem.php:265
/drone/src/lib/private/Encryption/EncryptionWrapper.php:89
/drone/src/tests/lib/Encryption/EncryptionWrapperTest.php:75
</details>


<details>
<summary>
2) Test\Files\External\PersonalMountTest::testFindByStorageId
</summary>

Your test case is not allowed to access the database.

/drone/src/tests/lib/TestCase.php:110
/drone/src/lib/composer/pimple/pimple/src/Pimple/Container.php:113
/drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:111
/drone/src/lib/private/ServerContainer.php:87
/drone/src/lib/private/Server.php:1141
/drone/src/lib/private/AllConfig.php:90
/drone/src/lib/private/AllConfig.php:340
/drone/src/lib/private/AllConfig.php:251
/drone/src/lib/private/legacy/util.php:233
/drone/src/lib/private/Files/Mount/Manager.php:124
/drone/src/tests/lib/Files/External/PersonalMountTest.php:47
</details>


<details>
<summary>
3) Test\Files\Mount\ManagerTest::testFind
</summary>

Your test case is not allowed to access the database.

/drone/src/tests/lib/TestCase.php:110
/drone/src/lib/composer/pimple/pimple/src/Pimple/Container.php:113
/drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:111
/drone/src/lib/private/ServerContainer.php:87
/drone/src/lib/private/Server.php:1141
/drone/src/lib/private/AllConfig.php:90
/drone/src/lib/private/AllConfig.php:340
/drone/src/lib/private/AllConfig.php:251
/drone/src/lib/private/legacy/util.php:233
/drone/src/lib/private/Files/Mount/Manager.php:72
/drone/src/tests/lib/Files/Mount/ManagerTest.php:31

</details>

<details>
<summary>
4) Test\Files\Mount\ManagerTest::testLong
</summary>


Your test case is not allowed to access the database.

/drone/src/tests/lib/TestCase.php:110
/drone/src/lib/composer/pimple/pimple/src/Pimple/Container.php:113
/drone/src/lib/private/AppFramework/Utility/SimpleContainer.php:111
/drone/src/lib/private/ServerContainer.php:87
/drone/src/lib/private/Server.php:1141
/drone/src/lib/private/AllConfig.php:90
/drone/src/lib/private/AllConfig.php:340
/drone/src/lib/private/AllConfig.php:251
/drone/src/lib/private/legacy/util.php:233
/drone/src/lib/private/Files/Mount/Manager.php:124
/drone/src/tests/lib/Files/Mount/ManagerTest.php:64

</details>

**Comment by DeepDiver1975** [comment]
> Some failures in drone after last commit:

I'd say there are some annotations missing then

**Comment by SergioBertolinSG** [comment]
Should I add 

* @group DB

to all those tests?

**Comment by SergioBertolinSG** [comment]
@DeepDiver1975  Can this be merged?

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 Can this be merged?

no - the code coverage report is 💩 - the expectation is that the coverage increases when adding mysql - not to decrease

**Comment by SergioBertolinSG** [comment]
Adding this commit https://github.com/owncloud/core/pull/29410/commits/f9500dc1257dc4e25bea74f41566535a16d2d7ee Reduced the coverage by 20,99%

Adding this one https://github.com/owncloud/core/pull/29410/commits/849fcf7d96f664d79c90724e1cc2a7893ebb6c8a incremented it 18,45%

Should it really appear incremented?

File: .drone.yml [hunk 2]
```diff
@@ -80,14 +80,27 @@ pipeline:
       matrix:
         TEST_SUITE: coverage
 
-  codecov:
+  codecov_sqlite:
```
Comment (DeepDiver1975) [inline .drone.yml]: one step should be enough - the plugin can take multiple files

Comment (DeepDiver1975) [reply .drone.yml]: or even with wild cards

**Comment by SergioBertolinSG** [comment]
Hmm flags appear in the report, but the github check doesn't split them

screen shot 2017-11-08 at 08 58 48

Do we need a different approach here? Making steps more explicit separating databases like in 794e529ddf9dcac7e03fc52dca4521e193d927b1 ?

**Comment by SergioBertolinSG** [comment]
Removed annotations in unit tests.

**Comment by DeepDiver1975** [comment]
@SergioBertolinSG please enable debug - https://github.com/robertstettner/drone-codecov#configuration
maybe we get some insights on how the plugin is uploading the reports

**Comment by DeepDiver1975** [comment]
Codecov upload has issues afaik. See the logs. Master Branch is used.

**Comment by SergioBertolinSG** [comment]
Yes, adding it here for reference:

```
==> Uploading reports
    url: https://codecov.io
query: branch=master&commit=1f658a6c9808a5817cea716aacd9b36614be2e12&build=1064&build_url=&name=&tag=&slug=owncloud%2Fcore&yaml=.codecov.yml&service=&flags=unit_tests&pr=29410&job=
```


Here https://github.com/robertstettner/drone-codecov/blob/master/entrypoint.sh#L44-L45

the plugin uses DRONE_BRANCH. 

In drone docs it says   ```DRONE_BRANCH | commit branch```

 http://docs.drone.io/environment-reference/

But there is also a variable called DRONE_COMMIT_BRANCH there. Not sure if it is the same.

**Comment by DeepDiver1975** [comment]
> But there is also a variable called DRONE_COMMIT_BRANCH there. Not sure if it is the same.

@tboerger is there a way to see the actual env vars? THX

**Comment by SergioBertolinSG** [comment]
Both DRONE_BRANCH  and DRONE_COMMIT_BRANCH are the same : master.

This variable contains the current branch name:
```
DRONE_COMMIT_REFSPEC=drone-enabling-mysql-code-coverage:master
```

**Pull request closed without merging**